### PR TITLE
Implement a first version of parallelization for SDDP

### PR DESCRIPTION
**Pull request by frapac**

### CODE REVIEW

File: examples/parallel_sddp.jl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Says whether to display results or not
+
+# Return
+* `V::Array{PolyhedralFunction}`:
+    the collection of approximation of the bellman functions
+"""
+function psolve_sddp(model, params, V; n_parallel_pass=4,
+                     synchronize=5, display=0)
+    # Redefine seeds in every processes to maximize randomness:
+    @everywhere srand()
```
Comment (leclere) [inline examples/parallel_sddp.jl:29]: Is this standard ? It feels wrong : the scenario choice on the different workers could be strongly related.

File: examples/parallel_sddp.jl [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+* `model::SPmodel`:
+    the stochastic problem we want to optimize
+* `param::SDDPparameters`:
+    the parameters of the SDDP algorithm
+* `V::Array{PolyhedralFunction}`:
+    the current estimation of Bellman's functions
+* `n_parallel_pass::Int`: default is 4
+    Number of parallel pass to compute
+* `synchronize::Int`: default is 5
+    Set when to synchronize the cuts between the different processes.
```
Comment (leclere) [inline examples/parallel_sddp.jl:18]: Unclear

File: examples/parallel_sddp.jl [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+# Arguments
+* `model::SPmodel`:
+    the stochastic problem we want to optimize
+* `param::SDDPparameters`:
+    the parameters of the SDDP algorithm
+* `V::Array{PolyhedralFunction}`:
+    the current estimation of Bellman's functions
+* `n_parallel_pass::Int`: default is 4
+    Number of parallel pass to compute
```
Comment (leclere) [inline examples/parallel_sddp.jl:16]: What is a parallel pass ?

File: examples/parallel_sddp.jl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* `V::Array{PolyhedralFunction}`:
+    the collection of approximation of the bellman functions
+"""
+function psolve_sddp(model, params, V; n_parallel_pass=4,
+                     synchronize=5, display=0)
+    # Redefine seeds in every processes to maximize randomness:
+    @everywhere srand()
+
+    mitn = params.maxItNumber
+    params.maxItNumber = synchronize
```
Comment (leclere) [inline examples/parallel_sddp.jl:32]: Maybe we could instead define a new params with something like :
one_thread_param = copy(params)
one_thread_param. maxItNumber = ...
etc.

**Comment by leclere** [comment]
I think that there is a lot of thing to test around the parallelization of sddp. We should try to discuss this at Buzios with other people who have implemented it.

File: examples/parallel_sddp.jl (hunk 1 above)
Comment (odow) [inline examples/parallel_sddp.jl:29]: You should do something like `srand(myid())` so the seed is different for each processor.

**Pull request closed without merging**